### ISSUE TITLE
Steve's Factory Manager Block Gates Can't Place Blocks

### ISSUE THREAD

**Issue by Cphoenix** [ForgeEssentials/ForgeEssentials#1773]
When FE permissions are enabled, Steve's Factory Manager block gates cannot place blocks in the world, they can however break blocks. Disabling FE fixes the issue. The problem occurs in all plots/zones and dimensions.

Furthermore, when SFM tries to place/break blocks, a new player in the permissions files is created called [SFM_PLAYER]. This player, while appearing in the list when using /p list users , is not actually editable, and trying to change its permissions, for example using /p user [SFM_PLAYER] allow blah.blah results in an error saying the player is unrecognised. Using /p list users after doing this will return two entries of [SFM_PLAYER] in the users list.

**Comment by luacs1998** [comment]
I will probably need to see how SFM handles its fake-players.

**Comment by Cphoenix** [comment]
You can see Vswe's source code here: https://github.com/Vswe/ModJam3

But I'm fairly certain that's not recent. Curse has a copy of the mod that was updated 16-March-2015, while the last commit on that repo is from last year. O.o

**Comment by luacs1998** [comment]
@vswe can I talk to you regarding this?

**Comment by luacs1998** [comment]
I am unable to recreate this in build 1002.

**Comment by Cphoenix** [comment]
Seems more-or-less fixed now, on build 1004. I first tried having the block gate place blocks on a plot owned by me. This didn't work, but the gate wasn't spewing items out either. I then did the same setup in the server zone, and it worked just fine. After that, I tried again on my own plot, and it's also working fine.

Wild guess here: Maybe FE didn't create the fake player file until SFM tried to do something on the server zone? I wasn't paying that close attention to what was happening.

Either way it works now.
